hibernate-core/src/main/java/org/hibernate/engine/spi/LoadQueryInfluencers.java

```diff
@@ -52,12 +52,19 @@
 
 	private final EffectiveEntityGraph effectiveEntityGraph = new EffectiveEntityGraph();
 
+	private Boolean readOnly;
```

Comment:
Isn't this really just a `boolean` with default `false`? Probably simplifies some `null` checks elsewhere.

Reply:
I chose Boolean type intentionally for sometimes (most of the times) its value is unknown and I wanna rely on some default value to take effect. That being said, I am gonna reconsider its design. Thanks for your comment.

Reply:
I think we cannot assume 'false' is the default value for 'readOnly'. For instance, **StatefulPersistenceContext** class has **setDefaultReadOnly**() method and it is likely the default value has been true already. Using Boolean type enjoys the flexibility to delay the decision at the last step to comply with the default value.
Also see QueryParameters's following method:
`
	public boolean isReadOnly(SharedSessionContractImplementor session) {
		return isReadOnlyInitialized
				? isReadOnly()
				: session.getPersistenceContextInternal().isDefaultReadOnly();
	}
`

Reply:
woa, you work on Sunday?